Sources/XcodeGenKit/SchemeGenerator.swift

```diff
@@ -228,6 +228,10 @@ public class SchemeGenerator {
         let buildableReference = buildActionEntries.first(where: { $0.buildableReference.blueprintName == schemeTarget?.name })?.buildableReference ?? buildActionEntries.first!.buildableReference
         let runnables = makeProductRunnables(for: schemeTarget, buildableReference: buildableReference)
 
+        let testMacroExpansion: XCScheme.BuildableReference = buildActionEntries.first(
+            where: { $0.buildableReference.blueprintName == scheme.test?.macroExpansion }
```

Comment:
Just in case `buildableReference.blueprintName` ever became nil and the where clause returned true when you hadn't set the macroExpansion, can you add a check that macroExpansion exists first (eg `if let macroExpansion = scheme.test?.macroExpansion {...`

Reply:
done! sorry, did a workaround for my project to unblock me and lost track of this PR, but this will allow me to do some cleanup on my code :D